mne/io/meas_info.py

```diff
@@ -41,13 +43,75 @@
     seeg=(FIFF.FIFFV_SEEG_CH, FIFF.FIFFV_COIL_NONE, FIFF.FIFF_UNIT_V),
 )
 
-
 def _summarize_str(st):
     """Aux function"""
     return st[:56][::-1].split(',', 1)[-1][::-1] + ', ...'
 
 
-class Info(dict):
+class _ChannelNameList(collections.Sequence):
+    """A read-only list, used to provide convenient acces to channel names."""
+    def __init__(self, info):
+        self._channels = info['chs']
+
+    def __getitem__(self, index):
+        if isinstance(index, int):
+            return self._channels[index]['ch_name']
+        elif isinstance(index, slice):
+            return list([ch['ch_name'] for ch in self._channels[index]])
+
+    def __len__(self):
+        return len(self._channels)
+
+    def __eq__(self, other):
+        return self.to_list() == other
+
+    def __repr__(self):
+        if len(self) == 0:
+            channels = 'empty'
+        elif len(self) < 10:
+            channels = ', '.join(self.to_list())
+        else:
+            channels = ', '.join(self[:5]) + ' ... ' + ', '.join(self[-5:])
+
+        return '<ChannelNameList | %d channels | %s>' % (len(self), channels)
+
+    def make_index_map(self):
+        """Construct a dictionary mapping channel names to integer indices
+
+        Use this to speed things up when lots of lookups are required, such as:
+        `info['ch_names'].index(ch)` or `ch in info['ch_names']`
+
+        Returns
+        -------
+        ch_name_to_index : dict (str -> int)
```

Comment:
you mean the name `ch_name_to_index`? Is there a rule that it should be the same name as the variable used internally?